Sources/CollectionViewManager.swift

```diff
@@ -722,35 +736,35 @@ open class CollectionViewManager: NSObject {
             remove(in: collectionView)
         }
     }
-    
+
     // MARK: - Private
-    
+
     /// Special wrapper for more convenient collection view updates.
     private func perform(updates: (UICollectionView?) -> Void, completion: Completion?) {
         collectionView.performBatchUpdates({ [weak collectionView] in
                                                updates(collectionView)
                                            }, completion: completion)
     }
-    
-    /// Use this method to perform register and set indexes operations for all section items.
+
+    /// Use this method to perform register all section items.
     func registerSectionItems() {
-        for index in 0..<_sectionItems.count {
-            let sectionItem = _sectionItems[index]
-            sectionItem.index = index
-            register(sectionItem)
+        for index in 0..<sectionItemsProvider.numberOfSections {
+            sectionItemsProvider[index]?.reusableViewItems.forEach { viewItem in
+                register(viewItem)
+            }
         }
+        sectionItemsProvider.registerKnownReuseTypes(in: collectionView)
```

Comment:
renamed to `registerReuseTypes `. In the case of `default` mode `collectionViewManager` can register all cellItems, sectionItems and viewItems here but in the case of `lazy` mode it can not be done. Empty implementation removed

Why did you remove indices setting? 

It performs in `recalculateIndexes` function, that calls from `update(_ sectionItems)`